app/routes.py

```diff
@@ -1,2 +1,223 @@
-from flask import Blueprint
+from flask import request, Blueprint, Response, jsonify, make_response
+from app import db
+from app.models.task import Task
+# wave 5 
+from app.models.goal import Goal
 
+# wave #2 
+from sqlalchemy import asc, desc
+# wave 3 
+from datetime import datetime
+# wave 4 
+import requests
+from app import slack_token 
+
+
+tasks_bp = Blueprint("tasks", __name__, url_prefix="/tasks")
+# bp path for wave 5
+goals_bp = Blueprint("goals", __name__, url_prefix="/goals")
+
+@tasks_bp.route("", methods=["GET", "POST"])
+def all_tasks():
+    if request.method == "GET":
+        sort = request.args.get("sort")
+        if sort == "asc": 
+            tasks = Task.query.order_by(asc("title"))
+        elif sort == "desc":
+            tasks = Task.query.order_by(desc("title"))     
+        else:
+            tasks = Task.query.all()
+        
+        tasks_response =[]
+        for task in tasks:
+            tasks_response.append(task.to_json())
+        
+        return jsonify(tasks_response)
+    
+    elif request.method == "POST":
+        request_body = request.get_json()
+        # Create a Task: Invalid Task With Missing Data
+        if "title" not in request_body or "description" not in request_body or "completed_at" not in request_body:
+            return make_response ({"details": "Invalid data"}),400
+        else:
+            new_task = Task(title=request_body["title"],
+                    description=request_body["description"],
+                    completed_at=request_body["completed_at"])
+        db.session.add(new_task)
+        db.session.commit()
+
+        return make_response({"task": new_task.to_json()}), 201 
+
+
+
+@tasks_bp.route("/<task_id>", methods=["GET", "PUT", "DELETE"])
+def one_task_only(task_id):
+    task = Task.query.get(task_id)
+    
+    # GET/PUT/DELETE request to /tasks/1 when there are no matching tasks and get this response
+    if task == None:
+        return make_response("",404)
```

Comment:
We could use `get_or_404()` method here 
https://flask-sqlalchemy.palletsprojects.com/en/2.x/api/#flask_sqlalchemy.BaseQuery.get_or_404